core/src/main/java/co/cask/wrangler/statistics/AddressFinder.java

```diff
@@ -0,0 +1,147 @@
+  private String STREET_SUFFIX_REGEX = "^(Avenue|Lane|Road|Boulevard|Drive|Street|Ave|Parkway|Way|Circle|Plaza|Mall|Dr|Rd|Blvd|Ln|St)$";
+  private String STREET_PREFIX_REGEX = "^(North|South|East|West|north|south|east|west|NORTH|SOUTH|WEST|EAST|N|S|E|W)$";
+  private String UNIT_NAME_REGEX = "^(Room|ROOM|room|RM|Apt|APT|apt|Unit|UNIT|unit)$";
+
+
+  public AddressFinder() {
+    words = readFromCsv(DICT_FILE);
+  }
+
+
+  /**
+   * Check if the input is in valid US address format
+   * @param str
+   * @return
+   */
+  public boolean isUSAddress(String str) {
+
+    return matchKeyWords(str);
+  }
+
+  /**
+   * Match string input with regular expression pattern for street address
+   * @param str
+   * @return
+   */
+  private boolean matchKeyWords(String str) {
+    Pattern pattern = Pattern.compile(US_REGEX_PATTERN);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  public boolean isUSZipCode (String str) {
+    Pattern pattern = Pattern.compile(US_ZIP_REGEX);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  public boolean isUSState(String str) {
+    Pattern pattern = Pattern.compile(US_STATE_REGEX);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  public boolean isCity(String str) {
+    if (isNumber(str)) {
+      return false;
+    }
+    if (!words.contains(str)) {
+      return true;
+    }
+    else {
+      return false;
+    }
+  }
+
+  public boolean isSuffix(String str) {
+    Pattern pattern = Pattern.compile(STREET_SUFFIX_REGEX);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  public boolean isPrefix(String str) {
+    Pattern pattern = Pattern.compile(STREET_PREFIX_REGEX);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  public boolean isUnitName(String str) {
+    Pattern pattern = Pattern.compile(UNIT_NAME_REGEX);
+    Matcher matcher = pattern.matcher(str);
+    return matcher.matches();
+  }
+
+  private Set<String> readFromCsv(String fileName) {
+    Set<String> set = new HashSet<>();
+    CSVReader reader = null;
+    try {
+      BufferedReader bReader = new BufferedReader(new InputStreamReader(
+              this.getClass().getResourceAsStream("/" + fileName)));
+      reader = new CSVReader(bReader);
+      String [] nextLine;
+      while ((nextLine = reader.readNext()) != null) {
+        for (String str : nextLine) {
+          set.add(str);
+        }
+      }
+      return set;
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
```

Comment:
Is there anything else we should do here? should it through a real exception? if not, we should at least log a good error message